src/planscape/planning/tests/test_planningarea_views.py

```diff
@@ -0,0 +1,917 @@
+            ["2010-11-01 00:01:01-05", self.scenario4_1.id],
+            ["2010-12-01 00:01:01-05", self.scenario4_2.id],
+            ["2010-04-03 00:01:01-05", self.scenario4_3.id],
+        ]
+        # using raw updates here, to override django's autoupdate of updated_at field
+        with connection.cursor() as cursor:
+            for so in scenario_update_overrides:
+                cursor.execute(
+                    "UPDATE planning_scenario SET updated_at = %s WHERE id = %s", so
+                )
+
+        planning_area_update_overrides = [
+            ["2010-01-01 00:01:01-05", self.planning_area1.id],
+            ["2010-02-01 00:01:01-05", self.planning_area2.id],
+            ["2010-03-01 00:01:01-05", self.planning_area3.id],
+            ["2010-04-01 00:01:01-05", self.planning_area4.id],
+            ["2010-11-01 00:01:01-05", self.planning_area5.id],
+            ["2010-06-01 00:01:01-05", self.planning_area6.id],
+        ]
+        # using raw updates here, to override django's autoupdate of updated_at field
+        with connection.cursor() as cursor:
+            for p in planning_area_update_overrides:
+                cursor.execute(
+                    "UPDATE planning_planningarea SET updated_at = %s WHERE id = %s", p
+                )
+
+        self.client.force_authenticate(self.user)
+        response = self.client.get(
+            reverse("planning:list_planning_areas"), {}, content_type="application/json"
+        )
+        planning_areas = json.loads(response.content)
+        updates_list = [(pa["name"], pa["latest_updated"]) for pa in planning_areas]
+        self.assertEqual(
+            updates_list,
+            [
+                ("test plan4", "2010-12-01T05:01:01Z"),
+                ("test plan5", "2010-11-01T05:01:01Z"),
+                ("test plan3", "2010-10-01T05:01:01Z"),
+                ("test plan1", "2010-09-01T05:01:01Z"),
+                ("test plan2", "2010-02-01T05:01:01Z"),
+            ],
+        )
+
+    def test_list_planning_areas_not_logged_in(self):
+        response = self.client.get(
+            reverse("planning:list_planning_areas"), {}, content_type="application/json"
+        )
+        self.assertEqual(response.status_code, 401)
+        self.assertJSONEqual(response.content, {"error": "Authentication Required"})
+
+    def test_list_planning_areas_empty_user(self):
+        self.client.force_authenticate(self.emptyuser)
+        response = self.client.get(
+            reverse("planning:list_planning_areas"), {}, content_type="application/json"
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.json()), 0)
+
+
+# EndtoEnd test that lists, creates a planning_area, creates a scenario,
+# tests what was stored, and then deletes everything.
+# This covers the basic happiest of cases and should not be a substitute
+# for the main unit tests.
+class EndtoEndPlanningAreaAndScenarioTest(APITransactionTestCase):
```

Comment:
should this be on a separate file as well?

Reply:
I was wondering the same thing, since it says it's an EndtoEnd test. But it's still basically just CRUD operations of a single PlanningArea entity, with just a few related records included, and no other components, so I decided it didn't qualify as a standalone test. Maybe it should be named a Lifecycle test, though?